modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/DynamicParametersTest.java

```diff
@@ -301,6 +704,29 @@ public Stream<DynamicTest> testFunction() {
         );
     }
 
+    /** Set operations. */
+    @TestFactory
+    public Stream<DynamicTest> testSetOps() {
+        return Stream.of(
+                sql("SELECT 1 UNION SELECT ?", 1)
+                        .parameterTypes(nullable(NativeTypes.INT32))
+                        .ok(),
+
+                sql("SELECT 1 UNION SELECT ?", Unspecified.UNKNOWN)
+                        .fails("Unable to determine type of a dynamic parameter#0."),
+
+                sql("SELECT ? UNION SELECT 1", 1)
+                        .parameterTypes(nullable(NativeTypes.INT32))
+                        .ok(),
+
+                sql("SELECT ? UNION SELECT 1", Unspecified.UNKNOWN)
+                        .fails("Unable to determine type of a dynamic parameter#0."),
+
+                sql("SELECT ? UNION SELECT ?", 1, Unspecified.UNKNOWN)
+                        .fails("Unable to determine type of a dynamic parameter#1.")
```

Comment:
Does it make sens to add smth like that
```
sql("SELECT 1 UNION SELECT ?", 1L)
sql("SELECT ? UNION SELECT ?", 1L, 2)
```

Reply:
You have a test that expects INT32 type, but here is a Long (INT64).
Will and should this work?

Reply:
Yes for both questions.

Reply:
>You have a test that expects INT32 type, but here is a Long (INT64).
Well, The correct answer to this question is the following:
- The test expect that the statement `SELECT ? UNION SELECT ?` works. if of one parameter is an integer and another is a bigint. Nothing more, nothing less.